From rpjday@crashcourse.ca Thu Mar 13 19:42:02 2008
From: "Robert P. J. Day" <rpjday@crashcourse.ca>
Date: Thu, 13 Mar 2008 22:41:52 -0400 (EDT)
Subject: SYSFS: Explicitly include required header file slab.h.
Cc: Greg KH <greg@kroah.com>
Message-ID: <alpine.LFD.1.00.0803132238310.30403@localhost.localdomain>



After an experimental deletion of the unnecessary inclusion of
<linux/slab.h> from the header file <linux/percpu.h>, the following
files under fs/sysfs were exposed as needing to explicitly include
<linux/slab.h>.

Signed-off-by: Robert P. J. Day <rpjday@crashcourse.ca>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/sysfs/dir.c  |    1 +
 fs/sysfs/file.c |    1 +
 2 files changed, 2 insertions(+)

--- a/fs/sysfs/dir.c
+++ b/fs/sysfs/dir.c
@@ -20,6 +20,7 @@
 #include <linux/idr.h>
 #include <linux/completion.h>
 #include <linux/mutex.h>
+#include <linux/slab.h>
 #include "sysfs.h"
 
 DEFINE_MUTEX(sysfs_mutex);
--- a/fs/sysfs/file.c
+++ b/fs/sysfs/file.c
@@ -13,6 +13,7 @@
 #include <linux/module.h>
 #include <linux/kobject.h>
 #include <linux/kallsyms.h>
+#include <linux/slab.h>
 #include <linux/namei.h>
 #include <linux/poll.h>
 #include <linux/list.h>
